src/lib/temple/front/kukai.ts

```diff
@@ -0,0 +1,86 @@
+import { entropyToMnemonic } from "bip39";
+import * as forge from "node-forge";
+import scrypt from "scryptsy";
+
+import { t } from "lib/i18n/react";
+
+export interface AccountCredentials {
+  sk: string | null;
+  pk: string | null;
+  pkh: string;
+  seedPhrase: string;
+}
+
+async function decrypt(chipher: string, password: string, salt: string) {
+  try {
+    if (!password || !salt) {
+      throw new Error("Missing password or salt");
+    }
+    const parts = chipher.split("==");
+    const chiphertext = parts[0];
+    const tag = parts[1];
+    const key = await scrypt.async(
+      password,
+      Buffer.from(salt, "hex"),
+      65536,
+      8,
+      1,
+      32
+    );
+    const decipher = forge.cipher.createDecipher(
+      "AES-GCM",
+      key.toString("binary")
+    );
+    decipher.start({
+      iv: Buffer.from(salt, "hex"),
+      tag: forge.util.createBuffer(
+        Buffer.from(tag, "hex").toString("binary"),
+        "utf-8"
+      ),
+    });
+    decipher.update(
+      forge.util.createBuffer(
+        Buffer.from(chiphertext, "hex").toString("binary"),
+        "utf-8"
+      )
+    );
+    const pass = decipher.finish();
+    if (pass) {
+      return Buffer.from(decipher.output.toHex(), "hex");
+    } else {
+      return null;
+    }
+  } catch (err) {
+    return null;
+  }
+}
+
+function bumpIV(salt: string, bumps: number) {
+  if (bumps > 255) {
```

Comment:
Why do we need `bumps`?   
It's passed in 1 place and used only in this check

Reply:
You should have asked Kukai developers but okay, it looks it can be removed